### PR TITLE
[BXMSPROD-1929] Centralize composite github actions

### PR DESCRIPTION
**Pull request by lampajr**
**Thank you for submitting this pull request**

**JIRA**: 

https://issues.redhat.com/browse/BXMSPROD-1929

**referenced Pull Requests**: 

Centralize composite GitHub actions here, merging those from droolsjbpm-build-bootstrap and kogito-pipelines.

Differences from the original ones:

* **build-chain**:
  * removed default `definition-file` as it depends on the final project structure, there is no default - this must be provided by any workflow using this action
* **maven**:
  * made `Setup Maven Settings` optional since it is not needed in droolsjbpm-build-boostrap related projects, they will have to explicitly set `setup-maven-settings` to "false".
* **dsl-tests**:
  * added new input `seed-repository` which defaults to `kiegroup/kogito-pipelines` but can be overridden to make the action much more generic. 

<pre>
How to retest a PR or trigger a specific build:

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
</pre>

### CODE REVIEW

**Comment by Ginxo** [comment]
related PRs for all the flows from repositories using either kiegroup/droolsjbpm-build-bootstrap or kiegroup/kogito-pipelines should be created once we agree on this. I know you know it @lampajr just for the record

**Comment by lampajr** [comment]
> related PRs for all the flows from repositories using either kiegroup/droolsjbpm-build-bootstrap or kiegroup/kogito-pipelines should be created once we agree on this. I know you know it @lampajr just for the record

Yeah exactly, they can be done at a later date once we agree on this

**Comment by radtriste** [comment]
For the `dsl-tests`, I would migrate it once and only if we decide to migrate the seed library of kogito-pipelines to here ... So that could be done later IMO
I put a point in my presentation tomorrow to discuss about this :)

**Review by radtriste — APPROVED** [review]
Really like it ! :)

File: .ci/actions/os-preparation/action.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        label_name: ${{ github.event.label.name }}
+        config_file: matrix_preparation/.github/matrix-os.json
+      shell: bash
+      run: |
+        os=$(jq --arg config_file "$config_file" --arg repo "$repo_name" --arg event "$event" --arg event_value "$label_name" 'map(.| select(.event==$event and ((.value==null and ((.repository | index( $repo))) or (.repository==null)) or .value==$event_value)))' $GITHUB_WORKSPACE/$config_file | jq "if (length > 1) then (.[] | select(.repository!=null)) else (.[] | select(.repository==null)) end | .os")
+        if [ -z "$os" ]; then 
+          echo "No configuration found for [repo=$repo_name, event=$event, value=$label_name] at $GITHUB_WORKSPACE/$config_file"
+          exit 1
+        else 
+          echo "::set-output name=operating_systems::$(echo $os)"
```
Comment (mareknovotny) [inline .ci/actions/os-preparation/action.yml:40]: this was already deprecated and fixed in https://issues.redhat.com/browse/BXMSPROD-1937

File: .ci/actions/parse-labels/action.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        filtered_labels="$(echo ${{ inputs.labels }} | jq -c 'map(select(.name | startswith("${{ inputs.label-prefix }}")))')"
+        echo "filtered_labels = ${filtered_labels}"
+        echo "FILTERED_LABELS=${filtered_labels}" >> $GITHUB_ENV
+
+    - name: Extract targets
+      id: extract-targets
+      shell: bash
+      run: |
+        targets="$(echo $FILTERED_LABELS | jq -c '[.[] | .name | sub("${{ inputs.label-prefix }}"; "")]')"
+        echo "::set-output name=targets::$(echo $targets)"
```
Comment (mareknovotny) [inline .ci/actions/parse-labels/action.yml:36]: again https://issues.redhat.com/browse/BXMSPROD-1937

File: .ci/actions/os-preparation/action.yml (hunk 1 above)
Comment (lampajr) [reply .ci/actions/os-preparation/action.yml:40]: you are right, I missed to propagate the change here. Doing it right now :)

**Comment by lampajr** [comment]
> For the `dsl-tests`, I would migrate it once and only if we decide to migrate the seed library of kogito-pipelines to here ... So that could be done later IMO I put a point in my presentation tomorrow to discuss about this :)

It makes sense, in this case I would remove it from this PR and as you said add it at later time when really needed (if so)

**Comment by lampajr** [comment]
As agreed I replicated changes performed here https://issues.redhat.com/browse/BXMSPROD-1937 and I removed `dsl-tests` just for now.

**Review by mareknovotny — APPROVED** [review]

**Comment by lampajr** [comment]
I propagated the change just merged here https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/2175, if you agree I would merge this one and make some other improvements (if needed) in separated PRs.

**Comment by mareknovotny** [comment]
sure, i waited for @Ginxo approval, but i guess it is just formal thing now ;)

**Pull request merged**